pkg/controller/machine.go

```diff
@@ -410,37 +410,41 @@ func (c *Controller) deleteMachineAndProviderInstance(prov cloud.Provider, provi
 }
 
 func (c *Controller) createInstanceForMachine(prov cloud.Provider, providerInstance instance.Instance, machine *machinev1alpha1.Machine, userdataProvider userdata.Provider, providerConfig *providerconfig.Config) error {
+	// case 1: validate the machine spec before getting the instance from cloud provider.
+	// even though this is a little bit premature and inefficient, it helps us detect invalid specification
+	{
```

Comment:
What's the advantage of putting this into a dedicated block?

Reply:
some variables are invisible outside of this block and their lifetime is limited also to this block.

Reply:
Ah okay, I personally am not too fond of that, because my monitor has a limited width and having a block that is not strictly required takes away some of that width :P

Reply:
@alvaroaleman  glad to hear you have a monitor :) 
@alvaroaleman  no problem I can remove it.